pkg/kube/namespace/filter.go

```diff
@@ -60,25 +61,26 @@ func NewDiscoveryNamespacesFilter(
 	namespaces.AddEventHandler(controllers.EventHandler[*corev1.Namespace]{
 		AddFunc: func(ns *corev1.Namespace) {
 			f.lock.Lock()
-			defer f.lock.Unlock()
+			created := f.namespaceCreatedLocked(ns.ObjectMeta)
```

Comment:
rename to namespaceCreated

Reply:
this one is supposed to be called under the lock. though I suppose we can just put the mutex handling inside the `namespaceCreated` function